src/gen/utils.py

```diff
@@ -42,18 +42,18 @@ def _parse_first_json_snippet(snippet):
 
 	return json_parsed_string
 
-def _required_keys_exist(assessment_json):
+def _required_keys_exist(assessment_json, keys_to_check):
     """
     _required_keys_exist checks if required keys exist in the given assessment_json
     """    
-    keys_to_check = set(["similarity_assessment", "precision_assessment"])
+    keys_to_check = set(keys_to_check)
```

Comment:
Can it contain duplicates or is this an additional measure of enforcement?

Reply:
no duplicates! I changes the code to accept `set` directly.